mrmustard/lab_dev/states/states.py

```diff
@@ -230,6 +232,31 @@ def __init__(
         self._representation = Fock.from_function(fock_state, n=self.n, cutoffs=self.cutoffs)
 
 
+class Sauron(Ket):
+    r"""The `n`-th Sauron state is an approximation of the `n`-th Fock states using
+    a ring of `n+1` coherent states. The reference to the Lord of the Rings comes from
+    the approximation becoming perfect in the limit for the radius of the ring going
+    to zero where vacuum (= darkness) is.
+    The formula for the Sauron state as a superposition of coherent states on a ring
+    is given in https://arxiv.org/abs/2305.17099:
+    .. math::
+        |\text{Sauron}(n)\rangle = \frac{1}{\mathcal{N}}\sum_{k=0}^{n} e^{i 2\pi k/(n+1)} |r e^{2\pi k/(n+1)}\rangle_c,
+    Args:
+        modes (Sequence[int]): The modes of the Sauron state.
+        n (int): The Fock state that is approximated.
+        r (float): The radius of the ring of coherent states, default is 0.1.
+    """
+
+    def __init__(self, modes, n=0, r=0.1):
+        phases = np.linspace(0, 2 * np.pi * (1 - 1 / (n + 1)), n + 1)
+        cs = np.exp(1j * phases)
+        bs = (r * cs)[..., None]
+        As = np.zeros([n + 1, 1, 1], dtype="complex128")
```

Comment:
The calculation of As, bs and cs should probably be a triples.py function and instead init the representation with `.from_function`